### PR TITLE
add item filter cover

### PR DESCRIPTION
**Pull request by GlodBlock**
This stuff will be rewritten when cover supports NBT.

### CODE REVIEW

**Review by Glease — CHANGES_REQUESTED** [review]
Don't forget to also rename the various occasion of `model` in variable names to `mode`. 

The UI looks a bit too sparse. Maybe it could be made a bit shorter.

File: src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            super(aTileEntity, 176, 107, GT_Utility.intToStack(aCoverID));
+            this.side = aSide;
+            this.coverID = aCoverID;
+            this.coverVariable = aCoverVariable;
+
+            GT_GuiIconButton b;
+            b = new GT_GuiIconButton(this, 0, startX + spaceX*0, startY+spaceY*0, GT_GuiIcon.WHITELIST).setTooltipText(trans("125","White List"));
+            b = new GT_GuiIconButton(this, 1, startX + spaceX*1, startY+spaceY*0, GT_GuiIcon.BLACKLIST).setTooltipText(trans("124","Black List"));
+
+            itemFilterButtons = new GT_GuiFakeItemButton(this ,startX + spaceX*0, startY+spaceY*3, GT_GuiIcon.SLOT_GRAY);
```
Comment (Glease) [inline src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java:165]: Would be great if you can draw a line of text explaining what this slot means.

File: src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            return true;
+    }
+
+    @Override
+    public int onCoverScrewdriverclick(byte aSide, int aCoverID, int aCoverVariable, ICoverable aTileEntity, EntityPlayer aPlayer, float aX, float aY, float aZ) {
+        int model = aCoverVariable & 1;
+        if (model == 1) model = 0;
+        else model = 1;
+        if (model == 1){
+            GT_Utility.sendChatToPlayer(aPlayer, trans("124", "Black List Model"));
```
[the remark this answers is not in the document]
Comment (GlodBlock) [reply src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java:71]: my fault. it should be Mode

File: src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        }
+        else{
+            GT_Utility.sendChatToPlayer(aPlayer, trans("125", "White List Model"));
```
[the remark this answers is not in the document]
Comment (Prometheus0000) [reply src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java:74]: Black List->Blacklist. Same for white

File: src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+            ItemStack tStack = aPlayer.inventory.getCurrentItem();
+            if (tStack != null){
+                aCoverVariable = (stackToInt(tStack) << 1) + (aCoverVariable & 1);
+                aTileEntity.setCoverDataAtSide(aSide, aCoverVariable);
+                GT_Utility.sendChatToPlayer(aPlayer, trans("301", "Item Filter: ") + tStack.getDisplayName());
+            }
+            else{
+                aCoverVariable = aCoverVariable & 1;
+                aTileEntity.setCoverDataAtSide(aSide, aCoverVariable);
+                GT_Utility.sendChatToPlayer(aPlayer, trans("300", "Clear Filter!"));
```
Comment (Prometheus0000) [inline src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java:60]: This looks like it should be filter cleared, but I can't tell from just the code

**Comment by GlodBlock** [comment]
the typo has been fixed.
i can't shorten the UI because the height of the texture is fixed.

File: src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java [hunk 5]
```diff
@@ -159,16 +158,17 @@ public GUI(byte aSide, int aCoverID, int aCoverVariable, ICoverable aTileEntity)
             this.coverVariable = aCoverVariable;
 
             GT_GuiIconButton b;
-            b = new GT_GuiIconButton(this, 0, startX + spaceX*0, startY+spaceY*0, GT_GuiIcon.WHITELIST).setTooltipText(trans("125","White List"));
-            b = new GT_GuiIconButton(this, 1, startX + spaceX*1, startY+spaceY*0, GT_GuiIcon.BLACKLIST).setTooltipText(trans("124","Black List"));
+            b = new GT_GuiIconButton(this, 0, startX + spaceX*0, startY+spaceY*0, GT_GuiIcon.WHITELIST).setTooltipText(trans("125","Whitelist"));
```
Comment (Glease) [inline src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java:161]: In above code you referred to translation 125 as `Whitelist Mode`, and here it becomes `Whitelist`. This is not going to work very well. Please use different identifiers.

File: src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java [hunk 6]
```diff
@@ -57,23 +55,24 @@ public boolean onCoverRightclick(byte aSide, int aCoverID, int aCoverVariable, I
             else{
                 aCoverVariable = aCoverVariable & 1;
                 aTileEntity.setCoverDataAtSide(aSide, aCoverVariable);
-                GT_Utility.sendChatToPlayer(aPlayer, trans("300", "Clear Filter!"));
+                GT_Utility.sendChatToPlayer(aPlayer, trans("300", "Filter Cleared!"));
             }
+            GT_Utility.sendChatToPlayer(aPlayer, ""+aCoverVariable);
```
Comment (Glease) [inline src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java:60]: the result of stackToInt is unintelligible to an ordinary user. Best not send it. What are you trying to acheive here?

File: src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java [hunk 7]
```diff
@@ -57,23 +55,24 @@ public boolean onCoverRightclick(byte aSide, int aCoverID, int aCoverVariable, I
...
 
     @Override
     public int onCoverScrewdriverclick(byte aSide, int aCoverID, int aCoverVariable, ICoverable aTileEntity, EntityPlayer aPlayer, float aX, float aY, float aZ) {
-        int model = aCoverVariable & 1;
-        if (model == 1) model = 0;
-        else model = 1;
-        if (model == 1){
-            GT_Utility.sendChatToPlayer(aPlayer, trans("124", "Black List Model"));
+        int mode = aCoverVariable & 1;
+        if (mode == 1) mode = 0;
```
Comment (Glease) [inline src/main/java/gregtech/common/covers/GT_Cover_ItemFilter.java:67]: Preferably collapse the two `if (mode ==1)` if into one. The current code style is not very readably, but that may be just me.

**Review by Glease — CHANGES_REQUESTED** [review]

**Review by Glease — APPROVED** [review]

File: src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java [hunk 8]
```diff
@@ -584,6 +563,10 @@ public GT_MetaGenerated_Item_01() {
 
         ItemList.FluidFilter.set(addItem(669, "Fluid Filter Cover", "Set with Fluid Container to only accept one Fluid Type"));
         GregTech_API.registerCover(ItemList.FluidFilter.get(1L), TextureFactory.of(MACHINE_CASINGS[1][0], TextureFactory.of(OVERLAY_SHUTTER)), new GT_Cover_Fluidfilter());
+        ItemList.ItemFilter_Export.set(addItem(270,"Item Filter Cover (Export)", "Right click with Item to set filter (Only support Export Mode)"));
```
Comment (Prometheus0000) [inline src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java:566]: Right click with Item to set filter (Only support Export Mode)
Right click with an item to set filter (Only supports Export Mode)

File: src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java [hunk 9]
```diff
@@ -584,6 +563,10 @@ public GT_MetaGenerated_Item_01() {
...
+        GregTech_API.registerCover(ItemList.ItemFilter_Export.get(1L), TextureFactory.of(MACHINE_CASINGS[5][0], TextureFactory.of(OVERLAY_CONVEYOR)), new GT_Cover_ItemFilter(true));
+        ItemList.ItemFilter_Import.set(addItem(271,"Item Filter Cover (Import)", "Right click with Item to set filter (Only support Import Mode)"));
```
Comment (Prometheus0000) [inline src/main/java/gregtech/common/items/GT_MetaGenerated_Item_01.java:568]: Right click with Item to set filter (Only support Import Mode)
Right click with an item to set filter (Only supports Import Mode)

**Comment by Prometheus0000** [comment]
Dream, why did you merge this before they fixed the grammar problems? Now someone needs to do a separate PR.

**Comment by Dream-Master** [comment]
oh didnt see it sorry. Just do a another pr please @Prometheus0000 thanks

**Pull request merged**